src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -889,10 +910,36 @@ void AndroidDeviceControllerWrapper::OnICDRegistrationInfoRequired()
 void AndroidDeviceControllerWrapper::OnICDRegistrationComplete(chip::NodeId icdNodeId, uint32_t icdCounter)
 {
     chip::DeviceLayer::StackUnlock unlock;
+
```

Comment:
I think we can create one ICDRegistrationInfo struct, and put it as a member in AndroidDeviceControllerWrapper, and store clientInfo there.
When generating the symmetric key, we could also store that key in ICDRegistrationInfo, 
When commissioning completes, we may apply them in ICD storage and always notify upper application with these info.
Note: for android-chip-tool, we may or may not apply ICD storage and further check-in message, therefore we may introduce the boolean variable to enable/disable the storage?

Reply:
When ICDRegistrationComplete was added to the current structure, passing symmetricKey was added. However, if commissioning fails, this key value is invalid, so erasing it should be considered when implementing the upper layer. And, I think it is appropriate to generate keys randomly. If it is based upon FabricID / nodeID, there is a risk that the key value can be guessed.

Reply:
Let's follow-up the unique symmetric key discussion in the follow-up

Reply:
I think we can mimic 
https://github.com/project-chip/connectedhomeip/blob/371756bb6ad0a2fe12464ef50f64629f2313b250/examples/chip-tool/commands/pairing/PairingCommand.cpp#L390 to remove the bad key and icd metadata when commmisioning complete in this wrapper file.

otherwises, you have to figure out how to implement this removal mechanism via java/kotlin API to this c++ wrapper?

Reply:
I already implemented using below code.
https://github.com/project-chip/connectedhomeip/pull/31193/files#diff-22864825009a652ecbadf8006425b968ec54dbbafd6f39c720a0c84f58c089f3R652

Reply:
thanks